app/Services/Backups/InitiateBackupService.php

```diff
@@ -111,6 +115,18 @@ public function handle(Server $server, string $name = null): Backup
             );
         }
 
+        // Check if the server has reached or exceeded it's backup limit
+        if (! $server->backup_limit || $server->backups()->where('is_successful', true)->count() >= $server->backup_limit) {
+            if($override) {
+                // Remove latest backup
+                $last_backup = $server->backups()->where('is_successful', true)->oldest()->first();
```

Comment:
Variables should be `$camelCase` and not `$snake_case`.

You also need to double-check that you don't have a null value returned here, since first can return a model, or a null value. I'd also prefer `where()->orderByDesc('created_at')->first()` so it is clear what we're ordering by, and intuitive to non-Laravel people what we're doing.

Reply:
About the null value I don't think its needed. I added a count check anyway.